node/src/reactor/main_reactor/reactor_state.rs

```diff
@@ -69,5 +71,8 @@ pub enum ReactorState {
     /// Node is currently caught up and is an active validator.
     Validate,
     /// Node should be shut down for upgrade.
-    ShutdownForUpgrade,
+    ShutdownForUpgrade {
+        /// Timestamp at which the node switched to this state.
+        time_switched: Timestamp,
```

Comment:
I would like to have this outside the enum variant, somewhere in the reactor, probably an `Option<Timestamp>`. We currently hold no state specific to the reactor mode of operation in the variants and I'd like to keep it consistent.

Reply:
Point taken, although I feel this is the most logical place to hold state specific to the reactor mode. If anything, I'd prefer moving values specific to particular modes here instead of the other way round. Let's see what others think as well.

Reply:
I would incline to not have this timestamp tracked here just to not complicate the status endpoint. You could maybe use the `last_progress` field of the reactor to track the timestamp when it switched to this state without adding additional fields.

Reply:
Hmm, but are we sure that nothing else would affect the `last_progress` field?

Reply:
I think we should try to not reuse fields like these which are crucial in the control logic. I would prefer an entirely new field altogether, especially if we can find a suitable name for it.

Reply:
Well I think the `last_progress` field is specifically designed to detect idleness in the reactor state. For each state it has a different meaning depending on what idleness means to that state. But @fizyk20 has a point here, maybe this field should not exist and each state should track its own idleness check using a field attached to that state.
@fizyk20 would adding this field to the `ShutdownForUpgrade` state affect the client also? It just feels weird to me to change the status endpoint in a patch release.

Reply:
I agree the current situation with the fields holding state is not ideal, but I am opposed to changing the variants right now, especially with the status endpoint problem. If we want to do this "properly", we need to refactor more than just this field into the specific states. I think of the solution proposed previously as a nice compromise.